src/main/java/edu/utdallas/davisbase/storage/IndexPage.java

```diff
@@ -0,0 +1,497 @@
+			file.seek(0x00);
+			file.writeByte(0);
+			int firstPageNo = AddLeafPage(file);
+			file.seek(0x05);
+			file.writeInt(firstPageNo); // writing root page no.
+			setPageasRoot(file, firstPageNo);
+		} catch (Exception e) {
+
+		}
+	}
+
+	private static void setPageasRoot(RandomAccessFile file, int pageNo) {
+		// TODO Auto-generated method stub
+		int seekParentByte = (pageNo - 1) * pageSize + 10;
+		try {
+			file.seek(seekParentByte);
+			file.writeInt(-1);// making root
+			updateMetaDataRoot(file, pageNo);
+		} catch (Exception e) {
+		}
+	}
+
+	private static void updateMetaDataRoot(RandomAccessFile file, int pageNo) throws IOException {
+		// TODO Auto-generated method stub
+		file.seek(metaDataRootPageNoOffsetInFile);
+		file.writeInt(pageNo);
+	}
+
+	// splitting leaf page
+	public static int splitLeafPage(RandomAccessFile file, int pageNo) throws IOException {
+		sortKeys(file, pageNo);
+		int newSiblingPageNo = AddLeafPage(file);
+		boolean rootflag = CheckifRootNode(file, pageNo);
+		int parentPageNo;
+		try {
+			if (rootflag) {
+				parentPageNo = AddInteriorPage(file);
+				setPageasRoot(file, parentPageNo);
+				setParent(file, pageNo, parentPageNo);
+			} else {
+				parentPageNo = getParent(file, pageNo);
+			}
+			setParent(file, newSiblingPageNo, parentPageNo);
+			splitLeafNodeData(file, pageNo, parentPageNo, newSiblingPageNo);
+			setParentRightChildPointer(file, parentPageNo, newSiblingPageNo);
+			setRightSiblingPointer(file, newSiblingPageNo, getRightSiblingPointer(file, pageNo));
+			setRightSiblingPointer(file, pageNo, newSiblingPageNo);
+
+		} catch (Exception e) {
+		}
+		return -1;
+	}
+
+	public static void setParent(RandomAccessFile file, int pageNo, int parentPageNo) {
+		long seekParentByte = convertPageNoToFileOffset(pageNo) + PAGE_OFFSET_OF_CELL_PARENT_PAGE_NO;
+		try {
+			file.seek(seekParentByte);
+			file.writeInt(parentPageNo);
+		} catch (Exception e) {
+
+		}
+	}
+
+	public static void splitInteriorPage(RandomAccessFile file, int pageNo) throws IOException {
+		sortKeys(file, pageNo);
+		int newSiblingPageNo = AddInteriorPage(file);
+		boolean rootflag = CheckifRootNode(file, pageNo);
+		int parentPageNo;
+		try {
+			if (rootflag) {
+				parentPageNo = AddInteriorPage(file);
+			} else {
+				parentPageNo = getParent(file, pageNo);
+			}
+			splitInteriorNodeData(file, pageNo, parentPageNo, newSiblingPageNo);
+			setParentRightChildPointer(file, parentPageNo, newSiblingPageNo);
+
+		} catch (Exception e) {
+
+		}
+	}
+
+	private static void splitInteriorNodeData(RandomAccessFile file, int pageNo, int parentPageNo, int siblingPageNo) {
+		// TODO make sure to update left child pointers
+		// TODO Auto-generated method stub
+
+		try {
+			long bigPageOffset = convertPageNoToFileOffset(pageNo);
+			file.seek(bigPageOffset + PAGE_OFFSET_OF_CELL_COUNT);
+			int noOfrecordsInBigPage = file.readShort();
+			int splitIndex = (int) Math.ceil(noOfrecordsInBigPage / 2); // this record goes into the parent.
```

Comment:
`Math.ceil(*)` doesn't work here since `noOfrecordsInBigPage / 2` is integer division, and so implicitly performs truncation (i.e. it doesn't preserve the remainder). At least one of the operands has to be a floating-point type in order to get floating-point division.

```suggestion
			int splitIndex = (int) Math.ceil(noOfrecordsInBigPage / 2.0); // this record goes into the parent.
```